packages/manager/apps/pci-private-network/src/pages/new/steps/ConfigurationStep.tsx

```diff
@@ -220,17 +219,31 @@ export default function ConfigurationStep({
   const missingNameError =
     isNetworkNameInputTouched && !store.form.privateNetworkName.length;
 
+  const isCidrIpAddressValid = isValidIpAddress(store.form.address);
+  const isCidrMaskValid = isValidCidrMask(store.form.cidr);
+
+  const privateNetworkIsEmpty = () => !store.form.privateNetworkName;
+  const createGatewayIsCheckAndProjectIsDiscovery = () =>
+    !store.form.createGateway && store.project.isDiscovery;
+  const dhcpIsCheckAndCidrNotValid = () =>
+    store.form.dhcp && (!isCidrIpAddressValid || !isCidrMaskValid);
+
+  const hadToDisabledCreateButton =
+    privateNetworkIsEmpty() ||
+    createGatewayIsCheckAndProjectIsDiscovery() ||
+    dhcpIsCheckAndCidrNotValid();
```

Comment:
I don't understand why you use functions here

Reply:
It's was just for clarity. Maybe I should've done like this

`
const privateNetworkIsEmpty = !store.form.privateNetworkName;
const createGatewayIsCheckAndProjectIsDiscovery = !store.form.createGateway && store.project.isDiscovery;
const dhcpIsCheckAndCidrNotValid = store.form.dhcp && (!isCidrIpAddressValid || !isCidrMaskValid);

const hadToDisabledCreateButton =
    privateNetworkIsEmpty ||
    createGatewayIsCheckAndProjectIsDiscovery ||
    dhcpIsCheckAndCidrNotValid
`

What do you think ?

Reply:
IMO, there is no need to use functions as you are handling predicates returned by store properties that can be assigned to simple contants.

Reply:
@Tsiorifamonjena I think this way is also clear, you could also use a useMemo to group it